feat: implement custom dashboard icons and annotation inheritance

dwmkerr opened the pull request:
- Add ark.mckinsey.com annotation inheritance from A2AServers to Agents and MCPServers to Tools  
- Implement custom icon support for dashboard resources using ark.mckinsey.com/dashboard-icon
- Chainsaw tests for annotation inheritance with test annotations
- Update dashboard UI components to support custom icons
- Add API support for exposing annotations in responses

<img width="858" height="546" alt="image" src="https://github.com/user-attachments/assets/383edf1b-65e5-4d1a-9456-f6da62f7c810" />

dwmkerr committed:
feat: implement custom dashboard icons and annotation inheritance

- Add ark.mckinsey.com annotation inheritance from A2AServers to Agents and MCPServers to Tools
- Implement custom icon support for dashboard resources using ark.mckinsey.com/dashboard-icon
- Add icon resolver utility and constants for annotation management
- Update dashboard components to use custom icons or fallback to default icons
- Add annotations field to API responses and TypeScript types
- Add comprehensive chainsaw tests for annotation inheritance
- Add A2A server row component for list view
- Fix ruff linting issues in ark-mcp service
- Update documentation for services and dashboard configuration

dwmkerr committed:
resolve: merge conflict in docs metadata

dwmkerr committed:
docs: update resource icons documentation

- Remove kubectl example section as requested
- Update to cover A2A servers, agents, MCP servers, and tools
- Remove bold formatting from bullet points
- Clarify that icons are optional with fallback behavior

skokaina commented on ark/internal/controller/a2aserver_controller.go:
```diff
@@ -202,18 +202,28 @@ func (r *A2AServerReconciler) buildAgentWithSkills(a2aServer *arkv1prealpha1.A2A
+	// Inherit ark.mckinsey.com annotations from A2AServer to Agent
+	// AAS-2657: Will replace with more idiomatic K8s spec.template pattern
+	for key, value := range a2aServer.Annotations {
+		if strings.HasPrefix(key, "ark.mckinsey.com/") {
```
we should use a constant for -> "ark.mckinsey.com/"

dwmkerr replied:
Excellent, good spot, I've done a through review of all annotations, applied consistent and more idiomatic naming, updated the constants / k8s controller / docs / api layer and dashboard

skokaina commented on ark/internal/controller/a2aserver_controller.go:
```diff
@@ -202,18 +202,28 @@ func (r *A2AServerReconciler) buildAgentWithSkills(a2aServer *arkv1prealpha1.A2A
 			Name:      agentName,
 			Namespace: a2aServer.Namespace,
 			Labels: map[string]string{
 				"a2a.server/name": a2aServer.Name,
```
contant for "a2a.server/name", it is used in different places, the linter should catch this normally

dwmkerr replied:
This has also been fixed and i've used the much more idiomatic `ark.mckinsey.com/a2a-server-name`

skokaina commented on docs/content/developer-guide/ark-dashboard.mdx:
```diff
@@ -0,0 +1,34 @@
+metadata:
+  name: weather-agent
+  annotations:
+    ark.mckinsey.com/dashboard-icon: "/icons/weather.svg"
```
we shouldn't know the path, it should be something like "ic-weather" and then we have a map that gets the path for the frontend or defaults to any default icon

dwmkerr replied:
I considered `ic-weather` and similar, but then realised we need to do some extra logic in the dashboard to parse it, differentiate between a data uri, 'real' uri and bundled URI. I think bundling is a little janky but if documented well still seems OK - see below [the actual image will show soon it just uses the path in `main` which is not available until merged]:

<img width="1049" height="810" alt="image" src="https://github.com/user-attachments/assets/d9a53381-2bb8-461e-b210-b8b2466f2e9f" />

Given we'd have to document a table of IDs -> resources it seems cleaner just to clearly note that there are bundled resources and not have the additional mapping, just keep the table up to date.

skokaina commented on docs/content/developer-guide/services.mdx:
```diff
@@ -46,68 +46,40 @@ Individual services can also be exposed using `kubectl port-forward` (as shown i
-| `ar.com/description` | Human-readable service description | `ARK Dashboard Web Interface` |
-| `ar.com/version` | Service version | `1.0.0` |
-
-### Discovery Integration
```
what is intentional to remove dicovery integration and service architecture sections from the services.mdx ?

dwmkerr replied:
Yes, I don't know where this came from but it is incorrect, these annotations don't exist [probably happened when someone was updating docs and claude/cursor was being 'helpful'] i've tried to carefully fix any doc I look at when I update a page

skokaina commented on services/ark-api-a2a/pyproject.toml:
```diff
@@ -65,4 +65,4 @@ exclude_lines = [
 
 [tool.uv.sources]
-ark-sdk = { path = "../../out/ark-sdk/py-sdk/dist/ark_sdk-0.1.25-py3-none-any.whl" }
+ark-sdk = { path = "../../out/ark-sdk/py-sdk/dist/ark_sdk-0.1.31-py3-none-any.whl" }
```
This should be dynamically updated by the build using version.txt, I have done it for the evaluator-llm and it works well, but been thinking of moving to using a direct path to the source, like

As a dependency:
[project]
  dependencies = [
      ...,
      "ark-sdk @ file://../ark-sdk-python",
  ]

 Or if the lib has pyproject.toml
  [tool.uv.sources]
  ark-sdk = { path = "../ark-sdk-python", editable = true }

skokaina replied:
Might not work as we use a builder.. however, using a sed in the deps or build target can still be good: ie 	
```shell
sed -i.bak 's|path = "../../out/ark-sdk/py-sdk/dist/ark_sdk-.*\.whl"|path = "../../out/ark-sdk/py-sdk/dist/ark_sdk-$(shell cat $(BUILD_ROOT)/version.txt)-py3-none-any.whl"|' pyproject.toml 
```

dwmkerr replied:
Would you be able to raise as a separate PR or put into the tech debt JIRA ticket? I actaully think this is good as well: 

https://github.com/mckinsey/agents-at-scale-ark/pull/13/files#diff-e1a2acbe21901197db3920c551b8ff0616d8d6c3017715d25d9c0362e2246441R10

skokaina commented on services/ark-api/ark-api/src/ark_api/api/v1/agents.py:
```diff
@@ -53,7 +55,7 @@ def agent_to_detail_response(agent: dict) -> AgentDetailResponse:
     annotations = metadata.get("annotations", {})
     
-    is_a2a = "a2a.server/address" in annotations
+    is_a2a = A2A_SERVER_ADDRESS_ANNOTATION in annotations
```
great !!

skokaina commented on services/ark-dashboard/ark-dashboard/lib/utils/icon-resolver.ts:
```diff
@@ -0,0 +1,16 @@
+  
+  return function CustomIcon({className}: {className?: string}) {
+    return React.createElement('img', {
+      src: iconAnnotation,
```
here we should map the iconAnnotation to a src we know and have in dic instead of passing directly the path, will prevent injections too

dwmkerr replied:
[added a note earlier]

dwmkerr committed:
refactor: centralize annotations and labels into dedicated packages

- Create ark/internal/annotations/ package with ARKPrefix constant and categorized annotations
- Create ark/internal/labels/ package for Kubernetes selection labels
- Update all controllers to use centralized constants instead of magic strings
- Fix inconsistent annotation patterns (a2a.server/* -> ark.mckinsey.com/a2a-server-*)
- Separate labels (short keys for selection) from annotations (metadata storage)
- Update TypeScript constants to match Go structure
- Update chainsaw tests and documentation to use consistent patterns
- Ensure Kubernetes RFC compliance with proper DNS subdomain format

dwmkerr committed:
refactor: update cross-project files for centralized annotations

- Update TypeScript constants to match Go annotation structure
- Update chainsaw tests to use ark.mckinsey.com annotation format
- Update samples documentation and Makefiles for new label patterns
- Update developer guide documentation for consistent annotation usage

dwmkerr committed:
refactor: complete annotation centralization across all services

- Add missing annotations (triggered-from, localhost-gateway-port) to Go constants
- Update fark CLI tool to use centralized annotation constants
- Update Python ark-api service to use centralized annotation constants
- Ensure consistent annotation usage across Go, Python, and TypeScript codebases

dwmkerr committed:
fix: update remaining annotation references for full consistency

- Add missing annotation constants (triggered-from, localhost-gateway-port)
- Update Python ark-api service to use centralized annotation constants
- Fix import statements for cross-package annotation usage
- Ensure all hardcoded annotation strings are replaced with constants

dwmkerr committed:
fix: complete Python service annotation constant migration

- Update ark_services.py to use imported annotation constants
- Remove remaining hardcoded annotation strings from Python codebase
- Ensure full consistency across Go, Python, and TypeScript services
